libs/verify-android-environment.sh

```diff
@@ -51,4 +51,40 @@ if [ -z "${ANDROID_NDK_TOOLCHAIN_DIR}" ]; then
   exit 1
 fi
 
+# Determine the Java command to use to start the JVM.
+if [ -n "$JAVA_HOME" ] ; then
```

Comment:
I wonder if instead we should just check `command -v java` first to see if `java` is already on the path, and fall back to using `$JAVA_HOME`. Doesn't really matter though, I guess.

Reply:
I actually think we should respect `JAVA_HOME` before the `java` command as this reads to me as "I want to use this specific Java distribution".

Reply:
Fwiw this is what gradle does: https://github.com/mozilla/application-services/blob/a20afc62b62adad6e94237f01c1ad9941a8d7b6c/gradlew#L69-L89